flybywiresim/aircraft
# feat(bleed): Ground air start

Fixes #[issue_no]

## Summary of Changes


This PR is implementing a basic Air Starter Unit to be able to start engines or supply the PACKs without the APU.

The most complicated part of this is the handling of the MSFS APU since that is still required for the engines to start.
In this case I decided to have the FADEC control when the MSFS APU is consuming fuel to deal with the fact that we need it to provide bleed pressure for the MSFS engine model.

Additionally while working on this PR, I had issues with the way we start the engines in the sim. Because of the lack of a real bleed system in MSFS engines start when they have fuel and there is a bleed source somewhere in the aircraft.

This leads to engines being able to start when no actual bleed air is reaching them.

I therefore masked the TOGGLE_STARTER#ID# event in the FADEC code and use the SET_STARTER_HELD event to actually command an engine start when all requirements are fulfilled. Since we have a custom bleed system I also made sure the engines are only able to start when they have enough bleed pressure. Should that pressure disappear before 25% N2 is reached, the engines will stop. 
I also replaced the fake starter valve indication logic on the ENG SD page with the real starter valve from the pneumatics simulation. In that process I also corrected the starter valve closing logic in the pneumatics sim to close it at 50% N2.

That part of the work can be seen as a jump off point for a proper EIU.



## Screenshots (if necessary)



![image](https://github.com/flybywiresim/aircraft/assets/5833783/1fc096d3-a6f2-48b5-a412-af0631acd1d2)
![image](https://github.com/flybywiresim/aircraft/assets/5833783/22e206be-79d5-4b6f-81e9-ba6463838352)




## References





## Additional context



Discord username (if different from GitHub): \_Chaoz_

## Testing instructions


1. Start C&D
2. Apply ground power to the aircraft
3. Align ADIRS
4. Switch the EFB on and open the Ground Services Page
5. Activate the Air starter unit and observe the BLEED SD page
6. Pressure should build on the Indication above the left engine 
7. Pack 1 should turn on as well
8. open the cross bleed valve by manually turning the X BLEED selector to OPEN
9. observe a pressure rise in the right system indication
10. Pack 2 should turn on
11. Turn on the left and right engine fuel pumps
12. Turn the ENG MODE SELECTOR to IGN/START 
13. Turn the ENG 2 MASTER to ON 
14. Eng 2 should startup normally
15. Turn ENG MODE SELECTOR to NORM
16. Turn off Eng 2
17. Turn X BLEED selector to SHUT
18. Repeat steps 12 and 13
21. Eng 2 should only start for a bit before stopping
22. Turn X BLEED selector to OPEN
23. Wait for pressure to build up again
24. Cycling ENG MASTER should make ENG 2 start
25. Turn X BLEED selector to SHUT while N2 is below 25%
26. ENG 2 should stop after system pressure falls to around 14 PSI
27. Turn X BLEED selector to OPEN
28. Wait for pressure to build up again
29. Cycling ENG MASTER should make ENG 2 start
30. Turn X BLEED selector to SHUT while N2 is at 25% or greater
31. ENG 2 should continue starting normally



## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
